### PR TITLE
Clean up downloadSegmentToLocal() method in SegmentFetcherAndLoader to prevent unnecessary sleep and wrong marker file

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
1. Remove all sleep
2. Do not create marker file after downloading
3. Simplify the log
4. Because exception is not handled in this method, directly throw it to the caller

### CODE REVIEW

File: pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java [hunk 1]
```diff
@@ -248,42 +248,33 @@ private boolean isNewSegmentMetadata(@Nonnull OfflineSegmentZKMetadata newSegmen
...
-      FileUtils.deleteQuietly(tempSegmentFile);
-      FileUtils.deleteQuietly(tempFile);
-      LOGGER.error("Caught exception downloading segment {} for table {}", segmentId, tableName, e);
-      Utils.rethrowException(e);
-      throw new AssertionError("Should not reach this");
+      FileUtils.moveDirectory(tempIndexDir, indexDir);
+      LOGGER.info("Successfully downloaded segment: {} for table: {}", segmentName, tableName);
+      return indexDir.getAbsolutePath();
+    } finally {
+      FileUtils.deleteQuietly(tempDir);
```
Comment (mcvsubbu) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java]: Please keep the log when exception happens.

File: pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java [hunk 2]
```diff
@@ -248,42 +248,33 @@ private boolean isNewSegmentMetadata(@Nonnull OfflineSegmentZKMetadata newSegmen
...
-      new File(segmentDir, "finishedLoading").createNewFile();
-      return segmentDir.getAbsolutePath();
-    } catch (Exception e) {
-      FileUtils.deleteQuietly(tempSegmentFile);
-      FileUtils.deleteQuietly(tempFile);
-      LOGGER.error("Caught exception downloading segment {} for table {}", segmentId, tableName, e);
-      Utils.rethrowException(e);
-      throw new AssertionError("Should not reach this");
+      FileUtils.moveDirectory(tempIndexDir, indexDir);
+      LOGGER.info("Successfully downloaded segment: {} for table: {}", segmentName, tableName);
```
Comment (mcvsubbu) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java]: please add the indexDir path as a last component of this log.

Comment (Jackie-Jiang) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java]: Added

File: pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java (hunk 1 above)
Comment (Jackie-Jiang) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java]: The caller will log the exception

**Pull request merged**